app/Actions/Photo/Archive.php

```diff
@@ -266,7 +275,10 @@ protected function zip(Collection $photos, string $variant): StreamedResponse
 						);
 					} while (array_key_exists($filename, $uniqueFilenames));
 				}
-				$zip->addFileFromStream($filename, $archiveFileInfo->getFile()->read());
+				$zipFileOption = new ZipFileOption();
+				$zipFileOption->setMethod($this->deflateLevel === -1 ? ZipMethod::STORE() : ZipMethod::DEFLATE());
+				$zipFileOption->setDeflateLevel($this->deflateLevel);
+				$zip->addFileFromStream($filename, $archiveFileInfo->getFile()->read(), $zipFileOption);
```

Comment:
Because we do not iterate over photos here. I have no access to `$photos->title` in this place.

I wanted to but then I noticed that `Photo/Archive` and `Album/Archive` use two totally different approaches. They also solve collisions of files names differently. :( The classes also contain duplicate code. :( In the long run we should clean up that, i.e. move the common code into a shared base class, ensure that collisions in filenames are solved consistently, etc.

But I did not feel like this should be part of this PR. This would probably imply to rewrite both classes completely :D